core/src/main/java/org/apache/brooklyn/util/core/config/ConfigBag.java

```diff
@@ -518,6 +574,46 @@ protected Object getStringKey(String key, boolean markUsed) {
     /**
      * @return Unresolved configuration value. May be overridden to provide resolution - @see {@link ResolvingConfigBag#getStringKeyMaybe(String, boolean)}
      */
+    protected synchronized Maybe<Object> getKeyMaybe(ConfigKey<?> key, boolean markUsed) {
```

Comment:
Yeah, not sure how to remove that without an almost-equal amount of code! The difference is it needs to call a `getRawStringKeyMaybe(name, markUsed)` or `getStringKeyMaybe(name, markUsed)`. Thinking of how we could achieve that:
* We could pass another boolean to say whether raw is desired. But I don't really like that pattern.
* There's not a nice java interface for passing two parameters, so we'd end up declaring our own interface to pass the different getter impl.

Do you think either of those is worth it @geomacy?